### PR TITLE
Fixing sign button color and replacing deprecated components in SignatureRequestOriginalWarning

### LINKED ISSUE #22171: [Bug]: Sign button is blue instead of danger for `eth_sign` requests
### Describe the bug

**Problem:** the Sign button for `eth_sign` type signatures is now blue instead of danger/red color.

### Expected behavior

I think it should be red/warning as it used to be, to indicate danger proceeding?

### Screenshots/Recordings

Now:
Screenshot from 2023-12-05 16-35-03

Before (i.e. 10.33.1):
Screenshot from 2023-12-05 16-37-35

### Steps to reproduce

1. Enable eth sign from settings
2. Go to test dapp
3. Trigger Eth Sign
4. Accept
5. See warning popover with blue Sign button

### Error messages or log output

_No response_

### Version

11.6.0 (some versions earlier too)

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**
This PR addresses a Sev2 visual regression bug that occurred due to incorrect migration of component API in a [previous PR](https://github.com/MetaMask/metamask-extension/pull/20070), which aimed to update stale components with component library components. Not only does this PR rectify these visual regressions, but it also extends the integration of the component library by replacing additional stale components.

## **Related issues**

Fixes: #22171

## **Manual testing steps**

To check component updates

1. Go to the latest [storybook build in this PR](https://output.circle-artifacts.com/output/job/91d2b3eb-5921-4431-83d6-daa89b0d6687/artifacts/0/storybook/index.html)
2. Search `SignatureRequestOriginalWarning` in the search bar 
3. Click on the result 
4. See updated `SignatureRequestOriginalWarning` modal component with correct buttons

To check component works in app

1. Enable eth sign from settings
2. Go to [test dapp](https://metamask.github.io/test-dapp/)
3. Trigger Eth Sign
4. Accept
5. See `SignatureRequestOriginalWarning` modal

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/9ad3e17a-624f-4fca-a089-b92b5c6b097b

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/e9d1f2a7-b335-40db-bd26-474d3978f169

Screencast below shows working in app

https://github.com/MetaMask/metamask-extension/assets/8112138/96504942-c412-4b7b-94da-d1c3fb5abe8b

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/app-components.scss [hunk 1]
```diff
@@ -62,7 +62,6 @@
 @import 'signature-request/index';
 @import 'signature-request-siwe/index';
 @import 'signature-request-original/index';
-@import 'signature-request-original/signature-request-original-warning/index';
```
Comment (georgewrmarshall) [inline ui/components/app/app-components.scss:65]: Removing import to deleted SCSS file

File: ui/components/app/signature-request-original/signature-request-original-warning/index.scss [hunk 2]
```diff
@@ -1,27 +0,0 @@
...
-      color: var(--color-text-default);
-    }
-  }
-
-  &__footer {
-    border-top: 1px solid var(--color-border-muted);
-    width: 100%;
-    height: 132px;
-  }
-}
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request-original/signature-request-original-warning/index.scss:27]: No longer need this CSS

File: ui/components/app/signature-request-original/signature-request-original-warning/signature-request-original-warning.js [hunk 3]
```diff
@@ -2,19 +2,33 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { useI18nContext } from '../../../../hooks/useI18nContext';
 
-import Popover from '../../../ui/popover';
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request-original/signature-request-original-warning/signature-request-original-warning.js:5]: This version of the Popover is deprecated

File: test/e2e/tests/eth-sign.spec.js [hunk 4]
```diff
@@ -74,7 +74,7 @@ describe('Eth sign', function () {
         await approveEthSign(
           driver,
           '[data-testid="page-container-footer-next"]',
-          '.signature-request-warning__footer__sign-button',
+          '[data-testid="signature-warning-sign-button"]',
```
Comment (georgewrmarshall) [inline test/e2e/tests/eth-sign.spec.js:77]: Replacing removed class name with data test id

File: test/e2e/tests/metrics/signature-approved.spec.js [hunk 5]
```diff
@@ -268,7 +268,7 @@ describe('Signature Approved Event @no-mmi', function () {
         await driver.delay(regularDelayMs);
         await driver.clickElement('[data-testid="page-container-footer-next"]');
         await driver.clickElement(
-          '.signature-request-warning__footer__sign-button',
+          '[data-testid="signature-warning-sign-button"]',
```
Comment (georgewrmarshall) [inline test/e2e/tests/metrics/signature-approved.spec.js:271]: Replacing removed class name with data test id

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM!

**Review by NidhiKJha — APPROVED** [review]

**Comment by georgewrmarshall** [comment]
Fixes missing overlay https://github.com/MetaMask/metamask-extension/pull/22756

**Pull request merged**